client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/channels/CoapChannel.java

```diff
@@ -0,0 +1,265 @@
+
+  public static final Logger LOG = LoggerFactory // NOSONAR
+          .getLogger(CoapChannel.class);
+
+  private static final Map<TransportType, ChannelDirection> SUPPORTED_TYPES = new HashMap<TransportType, ChannelDirection>();
+
+  static {
+    SUPPORTED_TYPES.put(TransportType.PROFILE, ChannelDirection.BIDIRECTIONAL);
+    SUPPORTED_TYPES.put(TransportType.CONFIGURATION, ChannelDirection.BIDIRECTIONAL);
+    SUPPORTED_TYPES.put(TransportType.NOTIFICATION, ChannelDirection.BIDIRECTIONAL);
+  }
+
+  private static final String CHANNEL_ID = "default_coap_channel";
+  private KaaDataDemultiplexer demultiplexer;
+  private KaaDataMultiplexer multiplexer;
+  //    private volatile State channelState = State.CLOSED;
+  private IpTransportInfo currentServer;
+
+  @Override
+  public void sync(TransportType type) {
+    sync(Collections.singleton(type));
+  }
+
+  @Override
+  public void sync(Set<TransportType> types) {
+
+    //        if (channelState == State.SHUTDOWN) {
+    //            LOG.info("Can't sync. Channel [{}] is down", getId());
+    //            return;
+    //        }
+    //        if (channelState == State.PAUSE) {
+    //            LOG.info("Can't sync. Channel [{}] is paused", getId());
+    //            return;
+    //        }
+    //        if (channelState != State.OPENED) {
+    //            LOG.info("Can't sync. Channel [{}] is waiting for CONNACK message + KAASYNC message", getId());
+    //            return;
+    //        }
+    if (multiplexer == null) {
+      LOG.warn("Can't sync. Channel {} multiplexer is not set", getId());
+      return;
+    }
+    if (demultiplexer == null) {
+      LOG.warn("Can't sync. Channel {} demultiplexer is not set", getId());
+      return;
+    }
+    //        if (currentServer == null || socket == null) {
+    //            LOG.warn("Can't sync. Server is {}, socket is \"{}\"", currentServer, socket);
+    //            return;
+    //        }
+
+    Map<TransportType, ChannelDirection> typeMap = new HashMap<>(getSupportedTransportTypes().size());
+    for (TransportType type : types) {
+      LOG.info("Processing sync {} for channel [{}]", type, getId());
+      ChannelDirection direction = getSupportedTransportTypes().get(type);
+      if (direction != null) {
+        typeMap.put(type, direction);
+      } else {
+        LOG.error("Unsupported type {} for channel [{}]", type, getId());
+      }
+      for (Map.Entry<TransportType, ChannelDirection> typeIt : getSupportedTransportTypes().entrySet()) {
+        if (!typeIt.getKey().equals(type)) {
+          typeMap.put(typeIt.getKey(), ChannelDirection.DOWN);
+        }
+      }
+    }
+    //        try {
+    //            sendKaaSyncRequest(typeMap);
+    //        } catch (Exception e) {
+    //            LOG.error("Failed to sync channel [{}]", getId(), e);
+    //        }
+
+
+  }
+
+  @Override
+  public void syncAll() {
+    //To change body of implemented methods use File | Settings | File Templates.
```

Comment:
Please, remove auto-generated stubs comments. Basically, we should try keeping all comments as informative as possible—reading stubs is not fun :slightly_smiling_face: